src/init.cpp

```diff
@@ -701,11 +701,21 @@ static void ThreadImport(std::vector<fs::path> vImportFiles)
     }
 
     // scan for better chains in the block chain database, that are not yet connected in the active best chain
-    BlockValidationState state;
-    if (!ActivateBestChain(state, chainparams)) {
-        LogPrintf("Failed to connect best block (%s)\n", FormatStateMessage(state));
-        StartShutdown();
-        return;
+
+    // XXX this is kind of a hack to avoid a lock inversion.
```

Comment:
in commit 1c173812bac34e263ae49e0bc80e45b437e95e8b:

Why is this a hack and what should be used instead of the hack? If there is nothing that can be used instead, this is not a hack and the comment should go.

Reply:
Fixed by cleaning up inline comments and adding documentation to the ChainstateManager pointer definitions.